docs/tutorials/HD143006_Part_2.py

```diff
@@ -0,0 +1,481 @@
+
+
+# Here we set up the tools that will allow us to visualize the results of the loop in TensorBoard.
+
+from mpol import (
+    losses,  # here MPoL loss functions will be used
+    connectors,  # required to calculate the residuals in log_figure
+)
+
+
+# setting up Writer to log values and images for display in TensorBoard
+from torch.utils.tensorboard import SummaryWriter
+import os
+
+
+logs_base_dir = "./logs/"
+writer = SummaryWriter(logs_base_dir)
+os.makedirs(logs_base_dir, exist_ok=True)
+# %load_ext tensorboard
+
+
+def log_figure(
+    model, residuals
+):  # this function takes a snapshot of the image state, imaged residuals, amplitude of model visibilities, and amplitude of residuals
+
+    # populate residual connector
+    residuals()
+
+    fig, ax = plt.subplots(ncols=2, nrows=2, figsize=(10, 10))
+    im = ax[0, 0].imshow(
+        np.squeeze(model.icube.sky_cube.detach().cpu().numpy()),  #
+        origin="lower",
+        interpolation="none",
+        extent=model.icube.coords.img_ext,
+    )
+    plt.colorbar(im, ax=ax[0, 0])
+
+    im = ax[0, 1].imshow(
+        np.squeeze(residuals.sky_cube.detach().cpu().numpy()),  #
+        origin="lower",
+        interpolation="none",
+        extent=residuals.coords.img_ext,
+    )
+    plt.colorbar(im, ax=ax[0, 1])
+
+    im = ax[1, 0].imshow(
+        np.squeeze(torch.log(model.fcube.ground_amp.detach()).cpu().numpy()),  #
+        origin="lower",
+        interpolation="none",
+        extent=residuals.coords.vis_ext,
+    )
+    plt.colorbar(im, ax=ax[1, 0])
+
+    im = ax[1, 1].imshow(
+        np.squeeze(torch.log(residuals.ground_amp.detach()).cpu().numpy()),  #
+        origin="lower",
+        interpolation="none",
+        extent=residuals.coords.vis_ext,
+    )
+    plt.colorbar(im, ax=ax[1, 1])
+
+    return fig
+
+
+# With these set up, we can now make our training function (instead of a loop, we use a function here since the training loop will be ran multiple times with different configurations). The hyperparameters, such as `epochs` and `lambda_TV`, are contained under `config`. Most of them are used in the loss functions and can be read about [here](../api.html#module-mpol.losses).
+
+
+def train(model, dataset, optimizer, config, writer=None, logevery=50):
```

Comment:
I'm a bit confused. Now you are introducing more priors than just the image positivity? I think you need to be more clear about what you're including, because earlier it seemed like you were saying that you were only using an image positivity prior.

Reply:
I agree with this, I just feel like I don't have the knowledge to explain each of the priors here. Adding the `lambda_sparsity` prior affects the sparsity of the image, the `lr` affects how aggressive the model learns, `epochs` determines the amount of iterations the model trains for, I can assume that `prior_intensity` affects the strangth of the `entropy` term, but I'm not quite sure what the prior `entropy` is. If you someone doesn't mind explaining this and confirming what I said about the other priors that would be very helpful.

Reply:
Sure, I'll give my thoughts. I actually think it's okay if you don't fully explain all of the priors here, but I think it would be useful to explicitly state that now you are expanding beyond the initial optimization shown above, which only required image positivity.

The lambda terms affect the strength of the regularizing terms, but don't affect the form of the regularizer. They are basically just coefficients to weight each term. So sparsity, total variation (TV), and entropy are all different priors/regularizers that we add to the total loss term. I'm working on explaining each of them in my paper, but I haven't written everything yet. I think for the purposes of the tutorial it would be fine to just mention that you're now using multiple priors, and direct the reader [to the API](https://mpol-dev.github.io/MPoL/api.html#module-mpol.losses) for more information on these losses.